wa/workloads/sysbench/__init__.py

```diff
@@ -0,0 +1,215 @@
+
+
+    See: https://github.com/akopytov/sysbench
+
+    """
+
+    parameters = [
+        Parameter('timeout', kind=int, default=300,
+                  description='''
+                  timeout for workload execution (adjust from default if
+                  running on a slow target and/or specifying a large value for
+                  ``max_requests``
+                  '''),
+        Parameter('test', kind=str, default='cpu',
+                  allowed_values=['fileio', 'cpu', 'memory', 'threads', 'mutex'],
+                  description='sysbench test to run'),
+        Parameter('num_threads', kind=int, default=8,
+                  description='''
+                  The number of threads sysbench will launch.
+                  '''),
+        Parameter('max_requests', kind=int, default=None,
+                  description='The limit for the total number of requests.'),
+        Parameter('max_time', kind=int, default=None,
+                  description='''
+                  The limit for the total execution time. If neither this nor
+                  ``max_requests`` is specified, this will default to 30
+                  seconds.
+                  '''),
+        Parameter('file_test_mode', default=None,
+                  allowed_values=['seqwr', 'seqrewr', 'seqrd', 'rndrd', 'rndwr', 'rndrw'],
+                  description='''
+                  File test mode to use. This should only be specified if
+                  ``test`` is ``"fileio"``; if that is the case and
+                  ``file_test_mode`` is not specified, it will default to
+                  ``"seqwr"`` (please see sysbench documentation for
+                  explanation of various modes).
+                  '''),
+        Parameter('cmd_params', kind=str, default='',
+                  description='''
+                  Additional parameters to be passed to sysbench as a single
+                  stiring.
+                  '''),
+        Parameter('taskset_mask', kind=int, default=0,
+                  description='''
+                  The processes spawned by sysbench will be pinned to cores as
+                  specified by this parameter.
+                  '''),
+    ]
+
+    def validate(self):
+        if (self.max_requests is None) and (self.max_time is None):
+            self.max_time = 30
+        if self.max_time and (self.max_time + 10) > self.timeout:
+            self.timeout = self.max_time + 10
+        if self.test == 'fileio' and not self.file_test_mode:
+            self.logger.debug('Test is "fileio" and no file_test_mode specified -- using default.')
+            self.file_test_mode = 'seqwr'
+        elif self.test != 'fileio' and self.file_test_mode:
+            raise ConfigError('file_test_mode must not be specified unless test is "fileio"')
+
+    @once
+    def initialize(self, context):
+        exe = Executable(self, self.target.abi, 'sysbench')
+        host_binary = context.resolver.get(exe, strict=False)
```

Comment:
Is there any reason for having `strict` set to `False`?

Reply:
Nope. Removed.